src/pages/ThemeProductsPage/index.tsx

```diff
@@ -0,0 +1,62 @@
+import { useParams } from 'react-router-dom';
+import * as S from './styles';
+import ThemeProductList from './ThemeProductList';
+import { useGetThemeInfo } from './useGetThemeInfo';
+import { useCallback, useRef } from 'react';
+import { useGetThemeProducts } from './useGetThemeProducts';
+
+const ThemeProductsPage = () => {
+  const { themeId } = useParams<{ themeId: string }>();
+  const id = Number(themeId);
+  const { themeInfo, error: themeError } = useGetThemeInfo(id);
+  const {
+    list: products,
+    isLoading,
+    error: productsError,
+    hasMore,
+    loadMore,
+  } = useGetThemeProducts(id, 10);
+
+  const observer = useRef<IntersectionObserver | null>(null);
+  const bottomRef = useCallback(
+    (node: HTMLDivElement | null) => {
+      if (isLoading) return;
+
+      observer.current?.disconnect();
+
+      observer.current = new IntersectionObserver(entries => {
+        if (entries[0].isIntersecting && hasMore) {
+          loadMore();
+        }
+      });
```

Comment:
```suggestion
      }, { rootMargin: '0px 0px 300px 0px'});
```

IntersectionObserver를 잘 활용하셨네요!
요렇게 rootMargin을 적용하시면 좀더 자연스러운 무한스크롤을 구현하여 사용자 경험이 더욱 좋아질 것 같아요~!